custom_components/bosch_alarm/config_flow.py

```diff
@@ -80,25 +98,39 @@ async def async_step_user(
                 step_id="user", data_schema=STEP_USER_DATA_SCHEMA
             )
 
-        errors = {}
+        (errors, model, serial_number) = await try_connect(self.hass, user_input)
 
-        try:
-            (model, serial_number) = await try_connect(self.hass, user_input)
+        if not errors:
             await self.async_set_unique_id(serial_number)
             self._abort_if_unique_id_configured()
             return self.async_create_entry(title="Bosch %s" % model, data=user_input)
-        except (OSError, ConnectionRefusedError, ssl.SSLError, asyncio.exceptions.TimeoutError):
-            errors["base"] = "cannot_connect"
-        except PermissionError:
-            errors["base"] = "invalid_auth"
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception("Unexpected exception")
-            errors["base"] = "unknown"
 
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_reauth(
```

Comment:
If we're already going to have multiple steps in the config flow, I think the best way to do this might be to have the first step only ask for host/port, get basicinfo without authentication, then, based on the panel model, decide whether we need to ask for the installer code or not.

Reply:
We don't have multiple steps, one step is used when setting it up initially, while the other is used when prompted for reauth. Though your right, it might make more sense to just do it with a seperate entire form that has more info on what the user needs

Reply:
Actually it appears i can even just reuse the same step instead of writing everything twice.